File: core/x/javaapi/codegen/src/test/resources/golden/expected/com/example/grts/Query.java
package com.example.grts;

import viaduct.engine.api.EngineObjectData;
import viaduct.engine.api.NodeReference;
import viaduct.engine.api.RootFieldReference;
import viaduct.java.api.context.ExecutionContext;
import viaduct.java.api.context.RootFieldCall;
import viaduct.java.api.globalid.GlobalID;
import viaduct.java.api.internal.InternalContext;
import viaduct.java.api.internal.NodeObjectBase;
import viaduct.java.api.internal.ObjectBase;
import viaduct.java.api.internal.OutputBuilderTypeChecker;
import viaduct.java.api.reflect.CompositeField;
import viaduct.java.api.reflect.Field;
import viaduct.java.api.reflect.RootObjectField;
import viaduct.java.api.reflect.Type;
import viaduct.java.api.reflect.TypeFields;
import viaduct.java.api.types.Arguments;
import java.time.Instant;
import java.time.LocalDate;
import java.time.OffsetTime;
import java.util.LinkedHashMap;
import java.util.function.Consumer;
import java.util.List;
import java.util.Map;

@SuppressWarnings("MissingOverride")
public class Query extends ObjectBase implements viaduct.java.api.types.Query {

    public static final Type<Query> Reflection = Type.ofClass(Query.class);

    public static final class Fields implements TypeFields<Query> {
        private Fields() {}

        public static final Field<Query> __typename =
                Field.of("__typename", Reflection);
                public static final RootObjectField<Query, Order, Query_Order_Arguments> order =
                                RootObjectField.of("order", Reflection, Order.Reflection, List.of("order"));

                public static final RootObjectField<Query, User, Arguments.NoArguments> topUser =
                                RootObjectField.of("topUser", Reflection, User.Reflection, List.of("topUser"));

                public static final CompositeField<Query, Order> popularOrders =
                                CompositeField.of("popularOrders", Reflection, Order.Reflection);

                public static final CompositeField<Query, User> trendingUsers =
                                CompositeField.of("trendingUsers", Reflection, User.Reflection);

                public static final RootObjectField<Query, OrderConnection, Query_OrdersConnection_Arguments> ordersConnection =
                                RootObjectField.of("ordersConnection", Reflection, OrderConnection.Reflection, List.of("ordersConnection"));

                public static final RootObjectField<Query, Order, Query_LookupOrder_Arguments> lookupOrder =
                                RootObjectField.of("lookupOrder", Reflection, Order.Reflection, List.of("lookupOrder"));

                public static final CompositeField<Query, Node> node =
                                CompositeField.of("node", Reflection, Node.Reflection);

                public static final CompositeField<Query, Node> nodes =
                                CompositeField.of("nodes", Reflection, Node.Reflection);

    }

        public static OrderRootFieldCall order() {
            return new OrderRootFieldCall();
        }

        /** Collects the arguments for a call to the order root field. */
        public static final class OrderRootFieldCall {
            private Consumer<Query_Order_Arguments.Builder> __arguments = __builder -> {};

            private OrderRootFieldCall() {}

            public OrderRootFieldCall id(String id) {
                __arguments = __arguments.andThen(__builder -> __builder.id(id));
                return this;
            }

            public RootFieldCall<Order> build() {
                return new RootFieldCall<Order>() {
                    @Override
                    public RootObjectField<?, Order, ? extends Arguments> field() {
                        return Fields.order;
                    }

                    @Override
                    public Arguments arguments(ExecutionContext context) {
                        Query_Order_Arguments.Builder __builder = Query_Order_Arguments.builder(context);
                        __arguments.accept(__builder);
                        return __builder.build();
                    }
                };
            }
        }

        public static RootFieldCall<User> topUser() {
            return new RootFieldCall<User>() {
                @Override
                public RootObjectField<?, User, ? extends Arguments> field() {
                    return Fields.topUser;
                }

                @Override
                public Arguments arguments(ExecutionContext context) {
                    return Arguments.None;
                }
            };
        }

        public static OrdersConnectionRootFieldCall ordersConnection() {
            return new OrdersConnectionRootFieldCall();
        }

        /** Collects the arguments for a call to the ordersConnection root field. */
        public static final class OrdersConnectionRootFieldCall {
            private Consumer<Query_OrdersConnection_Arguments.Builder> __arguments = __builder -> {};

            private OrdersConnectionRootFieldCall() {}

            public OrdersConnectionRootFieldCall first(Integer first) {
                __arguments = __arguments.andThen(__builder -> __builder.first(first));
                return this;
            }

            public RootFieldCall<OrderConnection> build() {
                return new RootFieldCall<OrderConnection>() {
                    @Override
                    public RootObjectField<?, OrderConnection, ? extends Arguments> field() {
                        return Fields.ordersConnection;
                    }

                    @Override
                    public Arguments arguments(ExecutionContext context) {
                        Query_OrdersConnection_Arguments.Builder __builder = Query_OrdersConnection_Arguments.builder(context);
                        __arguments.accept(__builder);
                        return __builder.build();
                    }
                };
            }
        }

        public static LookupOrderRootFieldCall lookupOrder() {
            return new LookupOrderRootFieldCall();
        }

        /** Collects the arguments for a call to the lookupOrder root field. */
        public static final class LookupOrderRootFieldCall {
            private Consumer<Query_LookupOrder_Arguments.Builder> __arguments = __builder -> {};

            private LookupOrderRootFieldCall() {}

            public LookupOrderRootFieldCall filter(OrderLookupInput filter) {
                __arguments = __arguments.andThen(__builder -> __builder.filter(filter));
                return this;
            }

            public RootFieldCall<Order> build() {
                return new RootFieldCall<Order>() {
                    @Override
                    public RootObjectField<?, Order, ? extends Arguments> field() {
                        return Fields.lookupOrder;
                    }

                    @Override
                    public Arguments arguments(ExecutionContext context) {
                        Query_LookupOrder_Arguments.Builder __builder = Query_LookupOrder_Arguments.builder(context);
                        __arguments.accept(__builder);
                        return __builder.build();
                    }
                };
            }
        }


    public Query(InternalContext context, EngineObjectData.Sync data) {
        super(context, data);
    }

    private Query(InternalContext context, Map<String, Object> data) {
        super(context, data);
    }

    public Query(InternalContext context, RootFieldReference rootFieldReference) {
        super(context, rootFieldReference);
    }
        public Order getOrderOrThrow() {
            return fetchObject("order", Order::new);
        }
        public Order getOrder() {
            return fetchObject("order", Order::new);
        }

        public User getTopUserOrThrow() {
            return fetchObject("topUser", User::new);
        }
        public User getTopUser() {
            return fetchObject("topUser", User::new);
        }

        public List<Order> getPopularOrdersOrThrow() {
            return fetchObjectList("popularOrders", Order::new);
        }
        public List<Order> getPopularOrders() {
            return fetchObjectList("popularOrders", Order::new);
        }

        public List<User> getTrendingUsersOrThrow() {
            return fetchObjectList("trendingUsers", User::new);
        }
        public List<User> getTrendingUsers() {
            return fetchObjectList("trendingUsers", User::new);
        }

        public OrderConnection getOrdersConnectionOrThrow() {
            return fetchObject("ordersConnection", OrderConnection::new);
        }
        public OrderConnection getOrdersConnection() {
            return fetchObject("ordersConnection", OrderConnection::new);
        }

        public Order getLookupOrderOrThrow() {
            return fetchObject("lookupOrder", Order::new);
        }
        public Order getLookupOrder() {
            return fetchObject("lookupOrder", Order::new);
        }

        public Node getNodeOrThrow() {
            return fetchAbstractObject("node", Node.class);
        }
        public Node getNode() {
            return fetchAbstractObject("node", Node.class);
        }

        public List<Node> getNodesOrThrow() {
            return fetchAbstractObjectList("nodes", Node.class);
        }
        public List<Node> getNodes() {
            return fetchAbstractObjectList("nodes", Node.class);
        }


    public static Builder builder(ExecutionContext context) {
        return new Builder(InternalContext.from(context));
    }

    public static class Builder {
        private final InternalContext __context;
        private final Map<String, Object> data = new LinkedHashMap<>();

        private Builder(InternalContext __context) {
            this.__context = __context;
        }

                public Builder order(Order order) {
                    order = OutputBuilderTypeChecker.checkField(
                            __context,
                            "Query",
                            "order",
                            Order.class,
                            order);
                    data.put("order", order);
        return this;
                }

                public Builder topUser(User topUser) {
                    topUser = OutputBuilderTypeChecker.checkField(
                            __context,
                            "Query",
                            "topUser",
                            User.class,
                            topUser);
                    data.put("topUser", topUser);
        return this;
                }

                public Builder popularOrders(List<Order> popularOrders) {
                    popularOrders = OutputBuilderTypeChecker.checkField(
                            __context,
                            "Query",
                            "popularOrders",
                            Order.class,
                            popularOrders);
                    data.put("popularOrders", popularOrders);
        return this;
                }

                public Builder trendingUsers(List<User> trendingUsers) {
                    trendingUsers = OutputBuilderTypeChecker.checkField(
                            __context,
                            "Query",
                            "trendingUsers",
                            User.class,
                            trendingUsers);
                    data.put("trendingUsers", trendingUsers);
        return this;
                }

                public Builder ordersConnection(OrderConnection ordersConnection) {
                    ordersConnection = OutputBuilderTypeChecker.checkField(
                            __context,
                            "Query",
                            "ordersConnection",
                            OrderConnection.class,
                            ordersConnection);
                    data.put("ordersConnection", ordersConnection);
        return this;
                }

                public Builder lookupOrder(Order lookupOrder) {
                    lookupOrder = OutputBuilderTypeChecker.checkField(
                            __context,
                            "Query",
                            "lookupOrder",
                            Order.class,
                            lookupOrder);
                    data.put("lookupOrder", lookupOrder);
        return this;
                }

                public Builder node(Node node) {
                    node = OutputBuilderTypeChecker.checkField(
                            __context,
                            "Query",
                            "node",
                            Node.class,
                            node);
                    data.put("node", node);
        return this;
                }

                public Builder nodes(List<Node> nodes) {
                    nodes = OutputBuilderTypeChecker.checkField(
                            __context,
                            "Query",
                            "nodes",
                            Node.class,
                            nodes);
                    data.put("nodes", nodes);
        return this;
                }


        public Query build() {
            return new Query(__context, new LinkedHashMap<>(data));
        }
    }
}

File: core/tenant/codegen/src/test/resources/golden/expected/grts/Query.kt
@file:Suppress("warnings")

package viaduct.api.grts

import viaduct.apiannotations.InternalApi
import viaduct.api.context.ExecutionContext
import viaduct.api.internal.InternalContext
import viaduct.api.internal.ObjectBase
import viaduct.engine.api.EngineObject
import viaduct.engine.api.EngineObjectData

@OptIn(InternalApi::class)
class Query(context: InternalContext, engineObject: EngineObject)
    : ObjectBase(context, engineObject), viaduct.api.types.Object,viaduct.api.types.Query
{
     fun getOrder(alias: String?): viaduct.api.grts.Order? = TODO()
     fun getOrder(): viaduct.api.grts.Order? = TODO()
     fun getOrderOrNull(alias: String?): viaduct.api.grts.Order? = TODO()
     fun getOrderOrNull(): viaduct.api.grts.Order? = TODO()

     fun getTopUser(alias: String?): viaduct.api.grts.User? = TODO()
     fun getTopUser(): viaduct.api.grts.User? = TODO()
     fun getTopUserOrNull(alias: String?): viaduct.api.grts.User? = TODO()
     fun getTopUserOrNull(): viaduct.api.grts.User? = TODO()

     fun getPopularOrders(alias: String?): kotlin.collections.List<viaduct.api.grts.Order> = TODO()
     fun getPopularOrders(): kotlin.collections.List<viaduct.api.grts.Order> = TODO()
     fun getPopularOrdersOrNull(alias: String?): kotlin.collections.List<viaduct.api.grts.Order>? = TODO()
     fun getPopularOrdersOrNull(): kotlin.collections.List<viaduct.api.grts.Order>? = TODO()

     fun getTrendingUsers(alias: String?): kotlin.collections.List<viaduct.api.grts.User> = TODO()
     fun getTrendingUsers(): kotlin.collections.List<viaduct.api.grts.User> = TODO()
     fun getTrendingUsersOrNull(alias: String?): kotlin.collections.List<viaduct.api.grts.User>? = TODO()
     fun getTrendingUsersOrNull(): kotlin.collections.List<viaduct.api.grts.User>? = TODO()

     fun getOrdersConnection(alias: String?): viaduct.api.grts.OrderConnection? = TODO()
     fun getOrdersConnection(): viaduct.api.grts.OrderConnection? = TODO()
     fun getOrdersConnectionOrNull(alias: String?): viaduct.api.grts.OrderConnection? = TODO()
     fun getOrdersConnectionOrNull(): viaduct.api.grts.OrderConnection? = TODO()

     fun getNode(alias: String?): viaduct.api.grts.Node? = TODO()
     fun getNode(): viaduct.api.grts.Node? = TODO()
     fun getNodeOrNull(alias: String?): viaduct.api.grts.Node? = TODO()
     fun getNodeOrNull(): viaduct.api.grts.Node? = TODO()

     fun getNodes(alias: String?): kotlin.collections.List<viaduct.api.grts.Node?> = TODO()
     fun getNodes(): kotlin.collections.List<viaduct.api.grts.Node?> = TODO()
     fun getNodesOrNull(alias: String?): kotlin.collections.List<viaduct.api.grts.Node?>? = TODO()
     fun getNodesOrNull(): kotlin.collections.List<viaduct.api.grts.Node?>? = TODO()


    fun toBuilder(): Builder =
        Builder(__context, __engineObject.type, toBuilderEOD())

    object of {
        operator fun invoke(context: ExecutionContext, block: Builder.() -> Unit): Query =
            Builder(context).apply(block).build()
    }

    class Builder : ObjectBase.Builder<Query> {
        constructor(context: ExecutionContext)
            : super(
                context as InternalContext,
                TODO() as graphql.schema.GraphQLObjectType,
                null
            )

        internal constructor(
            context: InternalContext,
            type: graphql.schema.GraphQLObjectType,
            baseEngineObjectData: EngineObjectData.Sync
        ) : super(context, type, baseEngineObjectData)

                  fun order(value: viaduct.api.grts.Order?): Builder = TODO()

                  fun topUser(value: viaduct.api.grts.User?): Builder = TODO()

                  fun popularOrders(value: kotlin.collections.List<viaduct.api.grts.Order>): Builder = TODO()

                  fun trendingUsers(value: kotlin.collections.List<viaduct.api.grts.User>): Builder = TODO()

                  fun ordersConnection(value: viaduct.api.grts.OrderConnection?): Builder = TODO()

                  fun node(value: viaduct.api.grts.Node?): Builder = TODO()

                  fun nodes(value: kotlin.collections.List<viaduct.api.grts.Node?>): Builder = TODO()


        final override fun build(): Query = TODO()
    }

    @OptIn(viaduct.apiannotations.InternalApi::class)
    object Reflection : viaduct.api.reflect.Type<viaduct.api.grts.Query> {
        override final val name = "Query"
        override final val kcls = viaduct.api.grts.Query::class
    }
    object Fields : viaduct.api.reflect.TypeFields<viaduct.api.grts.Query> {
            final val __typename: viaduct.api.reflect.Field<viaduct.api.grts.Query> =
                viaduct.api.internal.FieldImpl("__typename", viaduct.api.grts.Query.Reflection)

            final val order: viaduct.api.reflect.RootObjectField<viaduct.api.grts.Query, viaduct.api.grts.Order, viaduct.api.grts.Query_Order_Arguments> =
                viaduct.api.internal.RootObjectFieldImpl("order", viaduct.api.grts.Query.Reflection, viaduct.api.grts.Order.Reflection, listOf("order"))

            final val topUser: viaduct.api.reflect.RootObjectField<viaduct.api.grts.Query, viaduct.api.grts.User, viaduct.api.types.Arguments.NoArguments> =
                viaduct.api.internal.RootObjectFieldImpl("topUser", viaduct.api.grts.Query.Reflection, viaduct.api.grts.User.Reflection, listOf("topUser"))

            final val popularOrders: viaduct.api.reflect.CompositeField<viaduct.api.grts.Query, viaduct.api.grts.Order> =
                viaduct.api.internal.CompositeFieldImpl("popularOrders", viaduct.api.grts.Query.Reflection, viaduct.api.grts.Order.Reflection)

            final val trendingUsers: viaduct.api.reflect.CompositeField<viaduct.api.grts.Query, viaduct.api.grts.User> =
                viaduct.api.internal.CompositeFieldImpl("trendingUsers", viaduct.api.grts.Query.Reflection, viaduct.api.grts.User.Reflection)

            final val ordersConnection: viaduct.api.reflect.RootObjectField<viaduct.api.grts.Query, viaduct.api.grts.OrderConnection, viaduct.api.grts.Query_OrdersConnection_Arguments> =
                viaduct.api.internal.RootObjectFieldImpl("ordersConnection", viaduct.api.grts.Query.Reflection, viaduct.api.grts.OrderConnection.Reflection, listOf("ordersConnection"))

            final val node: viaduct.api.reflect.CompositeField<viaduct.api.grts.Query, viaduct.api.grts.Node> =
                viaduct.api.internal.CompositeFieldImpl("node", viaduct.api.grts.Query.Reflection, viaduct.api.grts.Node.Reflection)

            final val nodes: viaduct.api.reflect.CompositeField<viaduct.api.grts.Query, viaduct.api.grts.Node> =
                viaduct.api.internal.CompositeFieldImpl("nodes", viaduct.api.grts.Query.Reflection, viaduct.api.grts.Node.Reflection)

    }

}